charts/newrelic-infrastructure-beta/templates/_helpers.tpl

```diff
@@ -0,0 +1,234 @@
+*/}}
+{{- define "newrelic.licenseSecretKey" -}}
+{{ include "newrelic.licenseCustomSecretKey" . | default "licenseKey" }}
+{{- end -}}
+
+{{/*
+Returns nrStaging
+*/}}
+{{- define "newrelic.nrStaging" -}}
+{{- if .Values.global }}
+  {{- if .Values.global.nrStaging }}
+    {{- .Values.global.nrStaging -}}
+  {{- end -}}
+{{- else if .Values.nrStaging }}
+  {{- .Values.nrStaging -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Returns fargate
+*/}}
+{{- define "newrelic.fargate" -}}
+{{- if .Values.global }}
+  {{- if .Values.global.fargate }}
+    {{- .Values.global.fargate -}}
+  {{- end -}}
+{{- else if .Values.fargate }}
+  {{- .Values.fargate -}}
+{{- end -}}
+{{- end -}}
+
+{{/* controlPlane scraper config */}}
+{{- define "newrelic.controlPlane.scraperConfig" -}}
+controlPlane:
+  enabled: true
+  {{- if .Values.controlPlane.scraperConfig.controlPlane }}
+    {{- .Values.controlPlane.scraperConfig.controlPlane | toYaml | nindent 2 }}
+  {{- end }}
+  {{- range $key, $val := .Values.controlPlane.scraperConfig }}
+    {{- if ne $key "controlPlane" -}}
+    {{- nindent 0 $key }}: {{ $val | quote }}
+    {{- end }}
+  {{- end }}
+{{- end }}
+
+{{/* kubelet scraper config */}}
+{{- define "newrelic.kubelet.scraperConfig" -}}
+kubelet:
+  enabled: true
+  {{- if .Values.kubelet.scraperConfig.kubelet }}
+    {{- .Values.kubelet.scraperConfig.kubelet | toYaml | nindent 2 }}
+  {{- end }}
+  {{- range $key, $val := .Values.kubelet.scraperConfig }}
+    {{- if ne $key "kubelet" -}}
+    {{- nindent 0 $key }}: {{ $val | quote }}
+    {{- end }}
+  {{- end }}
+{{- end }}
+
+{{- define "newrelic.deprecatedKubeStateMetrics" -}}
+ksm:
+  scheme: {{  $.Values.kubeStateMetricsScheme | quote }}
+  port: {{  $.Values.kubeStateMetricsPort | quote }}
+  staticURL: {{  $.Values.kubeStateMetricsUrl | quote }}
+  selector: {{  $.Values.kubeStateMetricsPodLabel | quote }}
+  namespace: {{  $.Values.kubeStateMetricsNamespace | quote }}
+{{- end -}}
+
+{{/* ksm scraper config */}}
+{{- define "newrelic.ksm.scraperConfig" -}}
+ksm:
+  enabled: true
+  {{- if .Values.ksm.scraperConfig.ksm }}
+    {{- .Values.ksm.scraperConfig.ksm | toYaml | nindent 2 }}
+  {{- end }}
+  {{- range $key, $val := .Values.ksm.scraperConfig }}
+    {{- if ne $key "ksm" -}}
+    {{- nindent 0 $key }}: {{ $val | quote }}
+    {{- end }}
+  {{- end }}
+{{- end }}
+
+{{/*
+Returns the list of namespaces where secrets need to be accessed by the controlPlane Scraper to do mTLS Auth
+*/}}
+{{- define "newrelic.roleBindingNamespaces" -}}
+{{ $namespaceList := list }}
+{{- range $components := .Values.controlPlane.scraperConfig.controlPlane }}
+    {{- range $autodiscover := $components.autodiscover }}
+        {{- range $endpoint := $autodiscover.endpoints }}
+            {{- if and ($endpoint.auth) }}
+            {{- if $endpoint.auth.mtls }}
+            {{- if $endpoint.auth.mtls.secretName }}
+            {{- $namespace := $endpoint.auth.mtls.secretNamespace | default "default" -}}
```

Comment:
Just pinging @gsanchezgavier to check if this default makes sense. Do we allow empty namespaces in the config with this meaning?

Reply:
for some reason i was not notified by this ping 😞 , or perhaps it lost in the nested mails sorry.

yes we should we made the change that empty namespaces for secrets are not allowed anymore, and we agree that instead of having this logic (which was the old one) we are just going to left the mtls example comment , which is already there i think